src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -283,7 +283,13 @@ class ExposureSubmissionCoordinator: NSObject, ExposureSubmissionCoordinating, R
 				self?.model.exposureSubmissionService.getTemporaryExposureKeys { error in
 					isLoading(false)
 					if let error = error {
-						self?.showErrorAlert(for: error)
+						if case .notAuthorized = error {
+							self?.model.exposureSubmissionService.isSubmissionConsentGiven = false
```

Comment:
both cases do set ```isSubmissionConsentGiven = false``` - better move out of the if?
I also wonder why not use ```if error == .notAuthorized``` here? It's better readable in my opinion.

Reply:
I have moved the line out if the if/else. 
The second proposal won't work as we have `if case .notAuthorized` (`if case` syntax).

Reply:
```
self?.model.exposureSubmissionService.isSubmissionConsentGiven = false
if error == .notAuthorized {
	Log.info("ExposureSubmissionCoordinator: Submission consent reset to false after OS authorization was not given.")
} else {
	self?.showErrorAlert(for: error)
}
```
if case syntax isn't needed, there is no associated value inside that case?
We can have a short talk about that, if you want to.

Reply:
Refactored